docs/drash/v2.x/2_tutorials/3_servers/4_error_handling.md

```diff
@@ -0,0 +1,324 @@
+# Error Handling
+
+## Table of Contents
+
+- [Before You Get Started](#before-you-get-started)
+- [Folder Structure End State](#folder-structure-end-state)
+- [Steps](#steps)
+- [Verification](#verification)
+- [Further Reading](#further-reading)
+
+## Before You Get Started
+
+### Default Behavior
+
+Under the hood (in `Drash.Server`), Drash uses its default `Drash.ErrorHandler`
+class to handle all errors thrown during the request-resource-response
+lifecycle. When the error handler catches errors, it returns an error response
+with a proper HTTP error status code. However, it also returns the stack trace
+and this behavior might not fit well with your application's requirements. For
+example, it will return something like the following:
+
+```text
+Error: Method Not Allowed
+    at Server.<anonymous> (https://deno.land/x/drash/src/http/server.ts:238:17)
+    at async Server.#respond (https://deno.land/std/http/server.ts:298:18)
+```
+
+### Overriding the Default Behavior
+
+You may want to override the default behavior so that clients do not see a stack
+trace. For example, you can override the default behavior to return an error
+response like the following:
+
+```json
+{
+  "message": "HTTP method `POST` is not allowed on this resource."
+}
+```
+
+Overriding the default behavior is a very simple process. All you need to do is
+to plug in a custom error handler class to the `error_handler` server config (as
+shown in the Steps section on this page).
+
+All error handler classes passed into the `error_handler` server config must
+match the `Drash.Interfaces.IErrorHandler` contract. In other words, make sure
+your error handler class has the following `catch` method definition:
+
+```typescript
+public catch(
+   error: Error, // Notice this is Error -- all errors should be an extension of Error
+   request: Request, // Notice this is the native Request
+   response: Drash.Response // Notice this is the built-in Drash Response
+): void { ... }
+```
+
+A full example looks like:
+
+```typescript
+class MyErrorHandler {
```

Comment:
This needs to either implement the ErrorHandler interface or extend the ErrorHandler class.

Reply:
doneski: 438c6de